text/0023-stats-component.md

```diff
@@ -0,0 +1,345 @@
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Let's split this section in to three parts. 
+First we will talk about what a default stats component might looks like and what stats we want to generate. Then let's talk about future possibilities, namely what a default stats component might look like for high dimentional data and what custom stats might look like and what we need to do to get there.
+
+## Default Stats component:
+To start, Qri can create a default stats component on tabular (two dimentional) data. Because we want the cost of creating a stats component to be low, we only want to generate stats that can be calculated on a rolling basis.
+
+The different types of data that we can generate stats on are number (including integer and floating point), string, and boolean.
+
+We can generate different stats depending on the type of that column.
+
+Number:
+  valid - the number of cells in that column that contain valid numbers
+  error - the number of cells in that column that contain invalid numbers
+  missing - the number of cells in that column that do not contain data
+  min - the minimum number found
+  max - the maximum number found
+  avg - the average value
+
+String
+  valid - the number of cells in that column that contain valid strings
+  error - the number of cells in that column that contain invalid string
+  missing - the number of cells in that column that do not contain data 
+  minLength - the shortest length of a string in that column
+  maxLength - the longest length of a string in that column
+
+Boolean
+  valid - the number of cells in that column that contain valid boolean values
+  error - the number of cells in that column that contain invalid boolean values
+  missing - the number of cells in that column that do not contain data 
+  trueCount - the number of cells that contain a true value
+  falseCount - the number of cells that contain a false value
+
+The structure of the default stats component depends on the structure of the data itself. If each row of the dataset body is expressed as an array, the stats component will be an array of objects. If each row of the dataset body is expressed as an object, the stats component will be an object of objects. One important thing to note, we must have a structure with a schema that describes the title and the type in order to create stats.
+
```

Comment:
So "One important thing to note, we must have a structure with a schema that describes the title and the type in order to create stats."... I don't actually know if this is true, but it feels like it needs to be. How else would we know at the start exactly what we are looking for when calculating the stats?